src/api/api.ts

```diff
@@ -1,120 +1,217 @@
+import { NavigateFunction } from "react-router-dom";
 import { AddExpenseData, AddMonthData, LoginData, SignupData } from "../components/constans/types";
-import { calculateDateForEachMonth } from "../components/utils/utils";
-import { supabase } from "../supabaseClient";
+const PATH = import.meta.env.VITE_PATH;
 
-export const addUser = async (values:SignupData, toggleAlertSuccess: { (alert: string): void; (arg0: string): void; }, toggleAlertError: { (alert: string): void; (arg0: string): void; }) => {
-    const { name, email, password } = values;
-    const { data, error } = await supabase.auth.signUp({
-        email: email,
-        password: password,
-    })
-    if (error) throw error;
-    if (data && data.user) {
-        const { data:userData, error } = await supabase
-        .from('users')
-        .insert([
-          { id: data.user?.id, name, email }
-        ])
-        if (error != null) {
-            toggleAlertError('User already exist.')
-            throw error;
-        };
-        if (userData === null) toggleAlertSuccess('Success! Account created!');
-    }
+export const addUser = async (values:SignupData, toggleAlertSuccess: { (alert: string): void; (arg0: string): void; }, toggleAlertError: { (alert: string): void; (arg0: string): void; }, navigate: NavigateFunction) => {
+    try {
+        const response = await fetch(`${PATH}/signup`, {
```

Comment:
Just a hint, I think you can create a function that will do post requests and re-use it for making different post requests. This will end up in writing less code ;) 

I suggest it because I see that we have to specify method and headers everywhere as well as stringify the JSON body